### PR TITLE
[WIP] Added DC/OS overlay network packages and its dependencies to the installer.

### PR DESCRIPTION
**Pull request by asridharan**
Added two packages:
  * mesos-overlay
  * cni

Also made the following modifications to the installer:
  * Introduced user configurable parameters `dcos_overlay_enable` and
   `dcos_overlay_network`.
  * Modified the `dcos-config.yaml` template to enable DC/OS overlay
    specific Mesos modules when the user enables DC/OS overlay by
    setting `dcos_overlay_enable` parameter to "true".
  * Added generation of `overlay_config.json` to specify the overlay
    networks  specified by user using the `dcos_overlay_network`
    parameter.

### CODE REVIEW

File: gen/aws/templates/cloudformation.json [hunk 1]
```diff
@@ -71,10 +71,10 @@
         "stable": "ami-05bc0164"
       },
       "us-west-1": {
-        "stable": "ami-27553a47"
```
Comment (cmaloney) [inline gen/aws/templates/cloudformation.json]: Why are we changing these AMIs?

File: gen/calc.py [hunk 2]
```diff
@@ -86,6 +86,22 @@ def calculate_mesos_slave_modules_json(mesos_slave_modules):
     return json_multiline.replace('\n', '\n' + injected_indent)
 
 
+def calculate_mesos_slave_modules(dcos_overlay_enable):
```
Comment (cmaloney) [inline gen/calc.py]: This really breaks adding more downstream modules... Would it be reasonable to wait to land this until the modules dir stuff @karya0 has been working on lands? That makes this really simple to do.

File: gen/calc.py [hunk 3]
```diff
@@ -296,7 +366,18 @@ def calculate_exhibitor_static_ensemble(master_list):
         'ui_banner_header_content': 'null',
         'ui_banner_footer_content': 'null',
         'ui_banner_image_path': 'null',
-        'ui_banner_dismissible': 'null'
+        'ui_banner_dismissible': 'null',
+        'dcos_overlay_network': '{                    \
```
Comment (cmaloney) [inline gen/calc.py]: This doesn't seem particularly safe to default. Probably we want to just lay out in docs waht all needs to be set and how to set it. That this is not configurable at cluster runtime is likely to also be an issue.

As a code style thing, if we want the multi-line format here, should use a python multi-line comment `"""` and declare the value earlier, then just reference that declared value here.

File: packages/mesos/buildinfo.json [hunk 4]
```diff
@@ -3,9 +3,9 @@
   "docker": "dcos-builder",
   "single_source" : {
     "kind": "git",
-    "git": "https://github.com/mesosphere/mesos",
-    "ref": "76c87d55a78e8cbb380fc0c71388ccec53ed1dc8",
-    "ref_origin" : "dcos-1.7"
+    "git": "git@github.com:apache/mesos.git",
```
Comment (cmaloney) [inline packages/mesos/buildinfo.json]: can't land changing mesos like this here. We bump DC/OS Mesos after a Mesos release

File: gen/aws/templates/cloudformation.json (hunk 1 above)
Comment (asridharan) [inline gen/aws/templates/cloudformation.json]: For DC/OS overlay we need docker 1.9 and higher. The cloud formation template has an older CoreOS AMI that comes with Docker 1.7. So in order to test the overlay features with cloud formation, we needed a newer AMI. 

I will move this to a separate commit so that it can be discarded in case we plan to upgrade the AMI using a separate commit.

File: gen/calc.py (hunk 2 above)
Comment (asridharan) [inline gen/calc.py]: We need this in 1.8. So if the infrastructure for adding modules is ready in that time frame, we can discard this PR and just use that infrastructure. Will check with @karya0 on this.

File: packages/mesos/buildinfo.json (hunk 4 above)
Comment (asridharan) [inline packages/mesos/buildinfo.json]: Will Make this into a separate commit, and discard the commit once we are done testing here.

File: gen/aws/templates/cloudformation.json (hunk 1 above)
Comment (cmaloney) [inline gen/aws/templates/cloudformation.json]: We need to at the very least give notice to support that we're changing the base requirements if we're going to update our Docker requirement. We also need to get from support if upgrading docker will be a dealbreaker for any customers (We really don't want to strand customers on old DC/OS).

As far as upgrading CoreOS, we need to do it in all regions/AMIs if we're doing it. And do it for other providers than AWS. Also note we intentionally haven't bumped CoreOS in a while: https://dcosjira.atlassian.net/browse/DCOS-38. We need to resolve that issue before we can update / change the CoreOS version.

Comment (asridharan) [inline gen/aws/templates/cloudformation.json]: @spahl @cmaloney has valid points here. We probably should looking at updating the base AMI ASAP else this might become a blocker?

File: gen/calc.py (hunk 2 above)
Comment (asridharan) [inline gen/calc.py]: Had a chat with @karya0. Looks like he should be adding the `--modules_dir` flag shortly. Once its merged upstream, I will go ahead and restructure this PR to use the `--modules_dir` flag on the Master and the Agent.

File: gen/calc.py (hunk 3 above)
Comment (malnick) [inline gen/calc.py]: I echo @cmaloney and would add that it might be nice to have the parameters configurable, that way in the future we could configure subsets of the defaulted JSON (though this is purely an armchair observation, it would make validating the contents of the config more attainable as well if we had a complete data structure and not a string blob).

Comment (sargun) [inline gen/calc.py]: @cmaloney Why don't you think it's safe to default?

Comment (cmaloney) [inline gen/calc.py]: It states we own / customers don't use several IP spaces.

It also says `'dcos_overlay_enable': 'true'` which both doesn't seem right (and breaks standard json practice...)

Comment (sargun) [inline gen/calc.py]: "44.128.0.0/16" is a safe set of IPs.
We could choose a different set for VTEPs. Like from 198.18.0.0/15 (maybe 198.19.0/20?)
